### ISSUE TITLE
Add Semantics Javadocs & update README.md

### ISSUE THREAD

**Issue by sammefford** [marklogic/java-client-api#346]

**Comment by sammefford** [comment]
README.md updated with commits 0223fe7 and d166399

**Comment by sammefford** [comment]
Ajit & Kiran & Kim, would you be able to read through the whats new sections 3.0.2-3.0.4 in [README.md](https://github.com/marklogic/java-client-api/blob/develop/README.md) and javadocs for com.marklogic.client.semantics and make sure things make sense?

**Comment by jmckean17** [comment]
Butting in here…

Sam:
I would reword this phrase –
     Semantics API - GraphManager for CRUD of semantic graphs; SPARQLQueryManager for executing SPARQL update, select, describe, construct, and ask queries

To –
    Semantics API - GraphManager for CRUD of semantic graphs; SPARQLQueryManager for executing SPARQL queries (select, describe, construct, and ask) and using SPARQL update for data and graphs.

SPARQL Update is a separate language from SPARQL. You can’t do select, describe, construct, or ask queries with SPARQL Update, and you can’t do updates with SPARQL, so I think they should be separate in the description.

Thanks,
--Jane

**Comment by kcoleman-marklogic** [comment]
Apologies for being slow to get to this, Sam. One very minor correction, and then I'll throw in an unsolicited opinion for free. Fix first:

In the 3.0-3 section: "Search Extract - add support to SearchHandle (actually MatchDocumentSummary) for content extracted by extract-document-data options".That should be  "...by the extract-document-data option" (there's just one of 'em).

Second, I don't think you need to say "Many bug fixes" for every release. That's one of those "take it as read" things, IMO. There will always be many bug fixes. I get why you said it for 3.0-2 and I think it is OK there.

**Comment by sammefford** [comment]
Thanks Kim & Jane.  I've incorporated your changes in d5412a0.

**Comment by georgeajit** [comment]
Bug Track 35154 has the comments and has been marked for ship
